src/watcher.js

```diff
@@ -0,0 +1,38 @@
+/* @flow */
+import Watchpack from 'watchpack';
+import debounce from 'debounce';
+
+import {createLogger} from './util/logger';
+
+const log = createLogger(__filename);
+
+
+export default function onSourceChange(
+    {sourceDir, artifactsDir, onChange}: Object): Watchpack {
+  // TODO: For network disks, we would need to add {poll: true}.
+  const watcher = new Watchpack();
+
+  const executeImmediately = true;
+  watcher.on('change', debounce((filePath) => {
+    proxyFileChanges({artifactsDir, onChange, filePath});
+  }, 1000, executeImmediately));
+
+  log.debug(`Watching for file changes in ${sourceDir}`);
+  watcher.watch([], [sourceDir], Date.now());
+
+  // TODO: support windows See:
+  // http://stackoverflow.com/questions/10021373/what-is-the-windows-equivalent-of-process-onsigint-in-node-js
+  process.on('SIGINT', () => watcher.close());
+  return watcher;
+}
+
+
+export function proxyFileChanges({artifactsDir, onChange, filePath}: Object) {
+  if (filePath.indexOf(artifactsDir) === 0) {
+    log.debug(`Ignoring change to: ${filePath}`);
```

Comment:
It would be nice if the watched files could be filtered using the same file filter already used by the build command by default.

e.g. the build file filter currently ignores the hidden files (I mean files which start with a '.', like temp file wrong the editors or the .gitignore), but currently any changes on dotfiles generate an xpi rebuild even if they will not be included into the xpi.